rfc-177-edition-all-content.md

```diff
@@ -0,0 +1,147 @@
+---
+status: proposed
+implementation: proposed
+status_last_reviewed: 2024-09-09
+---
+
+# Make all content types editionable
+
+## Summary
+
+GOV.UK has a number of types of content published using Whitehall Publisher. Some of these are editionable (i.e. publishers can create a draft version of the document before publishing) whereas others are not editionable (i.e. changes become live immediately).
+
+This RFC proposes using the edition workflow for all publishable entities in Whitehall Publisher.
+
+## Problem
+
+Documents published outside the edition workflow suffer from the following user-facing issues:
+
+* no draft preview (whether for the publisher themselves, or using a shareable preview link for fact checking)
+* no [2i functionality](https://www.gov.uk/guidance/how-to-publish-on-gov-uk/reviewing-and-approving-content\#a-2i-checklist)
+* no scheduled publishing
+* high risk of content errors as changes go live immediately
+* no audit trail
+* validations are not consistent with other document types, e.g. the length of a document summary
+
+Additionally, there are a number of developer facing issues:
+
+* a high number of callbacks are manually implemented on models to republish associated non-editionable content when changes are made in other types of non-editionable content, which has previously led to incidents where callbacks are accidentally missed or called too frequently
+* two different sets of code exist for republishing the editionable and non-editionable content, with much code for the non-editionable code being specific to the content type; this affects the developer experiences, making it more difficult to make changes, due to hardcoded special cases, a lack of flexibility and developers needing to understand additional code
+
+The edition workflow offers the following features, that are not available on non-editionable models:
+
+* 2i is mandatory; unless the user has permissions to force publish
+* all documents can be scheduled to publish in advance
+* all documents have a shareable preview link, which can be used for fact checking by third parties without a Signon account
+* there is reduced risk of errors being published on website by giving publishers a draft preview
+* an audit trail is kept of all edits made (including the exact edit, when and by whom)
+* the workflow is uniform with other document types, so publishers will already be familiar with the process without needing to learn a new user interface specifically for the content type
+* documents can be searched and filtered, e.g. by world location
+* all documents can be found in a central location in Whitehall, negating the need for the ‘More’ option on the heading menu
```

Comment:
Although there are links to other things on the "More" page - cabinet ministers ordering, emergency banner settings, sitewide settings, and republishing content - but maybe we can come up with a more descriptive name for those!